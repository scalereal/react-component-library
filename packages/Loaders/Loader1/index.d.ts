import * as React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
export declare const SizeMap: Props['size'][];
declare const Loader1: React.FC<Props> & WithStyle;
export default Loader1;
