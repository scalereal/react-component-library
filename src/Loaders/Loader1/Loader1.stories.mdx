import Loader1 from '.';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';
import { SizeMap } from '../types';

<Meta
    title="Components/Loaders/Loader1"
    component={Loader1}
    argTypes={{
        size: {
            control: {
                type: 'select',
                options: Object.values(SizeMap)
            }
        },
        Color: {
            control: 'color'
        },
        display: {
            control: {
                type: 'boolean'
            }
        },
        ref: {
            table: {
                disable: true
            }
        },
        theme: {
            table: {
                disable: true
            }
        },
        as: {
            table: {
                disable: true
            }
        }
    }}
/>

# Loaders

Displays a Loader.

export const Template = args => (
    <div>
        <Loader1 {...args} />
    </div>
);

<Preview>
    <Story name="Loader1" args={{ size: 'S' }}>
        {Template.bind({})}
    </Story>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around' }}>
        <Loader1 size="S" Color="green" />
        <Loader1 size="S" Color="red" />
    </div>
</Preview>

### Sizes

There are Six sizs available `XXS` | `XS` | `S` | `M` | `L` | `XL`.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-between' }}>
        <DotsCircle size="XXS" />
        <Loader1 size="XS" />
        <Loader1 size="S" />
        <Loader1 size="M" />
        <Loader1 size="L" />
        <Loader1 size="XL" />
    </div>
</Preview>

### Props

<Props story="Loader1" />
