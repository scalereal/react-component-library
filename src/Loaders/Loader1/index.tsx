import React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
import { Loader, LoaderWrapper } from './styled';

export const SizeMap: Props['size'][] = ['XXS','XS', 'S', 'M', 'L', 'XL'];

const Loader1:React.FC<Props> & WithStyle = React.memo(React.forwardRef( (props, ref) => {
    const {size} = props;

    return (
        <LoaderWrapper ref={ref} size={size} {...props}>
            <Loader {...props}/>
        </LoaderWrapper>
    );
    
}));

Loader1.displayName="Loader1";
Loader1.defaultProps = {
    size:'S',
    display: true
};

export default Loader1;
