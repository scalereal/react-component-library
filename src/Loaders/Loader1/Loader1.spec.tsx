import { render } from '@testing-library/react';
import * as React from 'react';
import Loader1 from '.';

describe('Loader1 Loader', () => {
    it('should render with default theme', () => {
        const { container } = render(<Loader1/>);
        expect(container).toMatchSnapshot();
    });
    it('should render with all the props given', () => {
        const { container } = render(
            <Loader1
                size="M"
                Color='#126AFA'
            />
        );
        expect(container).toMatchSnapshot();
    });

    it('should render image avatar properly', () => {
        const { container } = render(
            <Loader1 size="L" />
        );
        expect(container).toMatchSnapshot();
    });

});