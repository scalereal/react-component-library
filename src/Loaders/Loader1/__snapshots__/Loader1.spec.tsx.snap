// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader1 Loader should render image avatar properly 1`] = `
<div>
  <div
    class="sc-gsTCUz gEOEBe"
  >
    <div
      class="sc-bdfBwQ hSFrdG"
    />
  </div>
</div>
`;

exports[`Loader1 Loader should render with all the props given 1`] = `
<div>
  <div
    class="sc-gsTCUz ijTmgU"
  >
    <div
      class="sc-bdfBwQ jNUwAr"
    />
  </div>
</div>
`;

exports[`Loader1 Loader should render with default theme 1`] = `
<div>
  <div
    class="sc-gsTCUz dkOYSx"
  >
    <div
      class="sc-bdfBwQ hSFrdG"
    />
  </div>
</div>
`;
